src/main/ApplicationImpl.cpp

```diff
@@ -333,6 +363,18 @@ ApplicationImpl::~ApplicationImpl()
     {
         mProcessManager->shutdown();
     }
+    if (!modeHasDatabase(mAppMode))
+    {
+        // Note: BucketManager::dropAll will delete the $BUCKETDIR/tmp which is
+        // where process and history manager write _their_ temp files to, so we
+        // make sure to tear them down first here.
+        mProcessManager = nullptr;
```

Comment:
should this be done after `joinAllThreads` below?

Reply:
oh, good catch. sure, fixed.